README.md

```diff
@@ -606,6 +606,7 @@ These functions provide a very clean way to build out very simple functions and
 | `filter` | <code>((a -> Boolean) &#124; Pred a) -> m a -> m a</code> |
 | `first` | `m (a -> b) -> m (Pair a c -> Pair b c)` |
 | `fold` | `Semigroup s => m s -> s` |
+| `fork` | `Async m => (a -> _) -> (b -> _) -> () -> m a b -> _` |
```

Comment:
Unless you were talking about the function that gets passed into the `fork` function?
For that it gets SUPER tricky as that cancel function is optional. If we make it required here things could get nasty really quick.

In that case, we *could* do something like, if the third argument is a function, go ahead and take one more that is the `Async`, otherwise call it done.

Now the alternative to this is to provide (2) functions `fork` and `forkCancel`. The `fork` function has your original definition, and the `forkCancel` takes your new definition. That way we are explicit about it and do not have to deal with the nasty.

What is your thoughts on that?